core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -1121,7 +1121,58 @@ public final Sql sql(String sql) {
     sql("select * from dept, lateral table(ramp(dept.deptno))").ok();
   }
 
-  @Test void testCollectionTableWithLateral2() {
+  @Test public void testTableFunctionInSelect() {
+    sql("select table_func(dept.deptno) as (f0, f1) from dept")
+        .conformance(SqlConformance.HIVE)
+        .ok();
+  }
+
+  @Test public void testTableFunctionInSelectWithUnion() {
+    sql("select * from ("
+        + "select a.deptno, table_func(a.deptno) as (f0, f1)"
```

Comment:
There is a few limitations in Hive[1].Other expressions are not allowed in SELECT.
`No other expressions are allowed in SELECT
SELECT pageid, explode(adid_list) AS myCol... is not supported`
[1] [https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF#LanguageManualUDF-explode(array)](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF#LanguageManualUDF-explode(array))

Reply:
Yes. But it was already discussed in https://github.com/apache/calcite/pull/1138/. 
![image](https://user-images.githubusercontent.com/37774589/89140153-6626de00-d573-11ea-99da-254385384fc1.png)